skbio/tree/_dists.py

```diff
@@ -0,0 +1,395 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from functools import partial
+from itertools import combinations
+
+import numpy as np
+
+from skbio.tree import TreeNode
+from skbio.stats.distance import DistanceMatrix
+from ._utils import _check_dist_metric, _check_shuffler
+
+
+def _check_ids(trees, ids):
+    """Check tree IDs."""
+    if ids is None:
+        return
+    if (n := len(ids)) != len(trees):
+        raise ValueError(f"ID number does not match tree number.")
+    if n > len(set(ids)):
+        raise ValueError(f"IDs contain duplicates.")
+
+
+def _shared(trees):
+    """Get shared taxa among multiple trees."""
+    taxon_sets = [t.subset() for t in trees]
+    shared = frozenset.intersection(*taxon_sets)
+    if not shared:
+        raise ValueError("No taxon is shared across all trees.")
```

Comment:
The trees will br shrinked to only shared taxa prior to the calculation of Robinson-Foulds distances. If the trees are empty, the RF distances will become 0, and the normalized RF distances will be 0 / 0 = 1. This is technically doable. I am not sure how justified it is though. Let me think...

Reply:
I implemented your suggestion. Specifically, if two trees share no taxon, RF will be 0 and normalized RF will be 1. In this way, any input trees will return values rather than throwing errors.